eth/rpc/backend/gas_used_test.go

```
@@ -0,0 +1,179 @@
+
+// TestGasUsedFunTokens verifies that gas used is correctly calculated for precompile "sendToBank" txs.
+// Test creates 3 txs: 2 successful and one failing.
+// Successful txs gas should be refunded and failing tx should consume 100% of the gas limit.
+// It also checks that txs are included in the same block and block gas is greater or equals
+// to the total gas used by txs.
+func (s *BackendSuite) TestGasUsedFunTokens() {
+	// Create funtoken from erc20
+	erc20Addr, err := eth.NewEIP55AddrFromStr(testContractAddress.String())
+	s.Require().NoError(err)
+
+	nonce, err := s.backend.GetTransactionCount(s.fundedAccEthAddr, rpc.EthPendingBlockNumber)
```

golangci-lint (1.62.2)

[error] ineffectual assignment to nonce

(ineffassign)